core/utils/profiler.hpp

```diff
@@ -5,30 +5,35 @@
 #include <chrono>
 
 class TicToc {
-  std::string name_;
+  std::string_view name_;
```

Comment:
May this be breaking change?
Should we add `TicTok(std::string &&) = delete; TicTok(std::string &) = delete;`